domainservice/src/main/kotlin/ua/marchenko/artauction/auction/service/kafka/AuctionCreatedEventKafkaProducer.kt

```diff
@@ -0,0 +1,36 @@
+package ua.marchenko.artauction.auction.service.kafka
+
+import org.apache.kafka.clients.producer.ProducerRecord
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+import reactor.core.publisher.Mono
+import reactor.kafka.sender.KafkaSender
+import reactor.kafka.sender.SenderRecord
+import reactor.kotlin.core.publisher.toMono
+import ua.marchenko.artauction.auction.mapper.toAuctionCreatedEventProto
+import ua.marchenko.artauction.auction.model.MongoAuction
+import ua.marchenko.internal.KafkaTopic
+
+@Component
+class AuctionCreatedEventKafkaProducer(private val kafkaSender: KafkaSender<String, ByteArray>) {
+
+    fun sendCreateAuctionEvent(auction: MongoAuction): Mono<Unit> {
+        val auctionEvent = auction.toAuctionCreatedEventProto()
+        return kafkaSender.send(
+            SenderRecord.create(
+                ProducerRecord(
+                    KafkaTopic.AuctionKafkaTopic.CREATED,
+                    auctionEvent.auction.artworkId,
+                    auctionEvent.toByteArray()
+                ),
+                null
+            ).toMono()
+        )
```

Comment:
```suggestion
        val auctionEvent = auction.toAuctionCreatedEventProto()
        return kafkaSender.send(auctionEvent.toRecord().toMono())
```
It looks like good place to use mapper or private method to improve readability